api/include/opentelemetry/metrics/meter_provider.h

```diff
@@ -20,15 +23,56 @@ class MeterProvider
 {
 public:
   virtual ~MeterProvider() = default;
+
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
   /**
    * Gets or creates a named Meter instance.
    *
-   * Optionally a version can be passed to create a named and versioned Meter
-   * instance.
+   * @since ABI_VERSION 2
+   *
+   * @param[in] name Meter instrumentation scope
+   * @param[in] version Instrumentation scope version, optional
+   * @param[in] schema_url Instrumentation scope schema URL, optional
+   * @param[in] attributes Instrumentation scope attributes, optional
+   */
+  virtual nostd::shared_ptr<Meter> GetMeter(
+      nostd::string_view name,
+      nostd::string_view version                 = "",
+      nostd::string_view schema_url              = "",
+      const common::KeyValueIterable *attributes = nullptr) noexcept = 0;
+#else
+  /**
+   * Gets or creates a named Meter instance.
+   *
+   * @since ABI_VERSION 1
+   *
+   * @param[in] name Meter instrumentation scope
+   * @param[in] version Instrumentation scope version, optional
+   * @param[in] schema_url Instrumentation scope schema URL, optional
    */
-  virtual nostd::shared_ptr<Meter> GetMeter(nostd::string_view library_name,
-                                            nostd::string_view library_version = "",
-                                            nostd::string_view schema_url      = "") noexcept = 0;
+  virtual nostd::shared_ptr<Meter> GetMeter(nostd::string_view name,
+                                            nostd::string_view version    = "",
+                                            nostd::string_view schema_url = "") noexcept = 0;
+#endif
+
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+  nostd::shared_ptr<Meter> GetMeter(
+      nostd::string_view name,
+      nostd::string_view version,
+      nostd::string_view schema_url,
+      std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>> attributes)
+  {
+    nostd::span<const std::pair<nostd::string_view, common::AttributeValue>> attributes_span{
+        attributes.begin(), attributes.end()};
+
+    common::KeyValueIterableView<
+        nostd::span<const std::pair<nostd::string_view, common::AttributeValue>>>
+        iterable_attributes{attributes_span};
+
+    return GetMeter(name, version, schema_url, &iterable_attributes);
+  }
```

Comment:
Not sure, but good to check if we need more overloads as done for [StartSpan](https://github.com/open-telemetry/opentelemetry-cpp/blob/61e7429d6362bd2dc166ca81f0cb28963eac0ee8/api/include/opentelemetry/trace/tracer.h#L50C1-L57C4) - to pass the instance of custom classes which are compliant to `is_key_value_iterable` trait?

Reply:
Fixed